### PR TITLE
fix(mention): No extra whitespace in markdown before/after mentions

### PR DESCRIPTION
**Pull request by mejo-**
Fixes: #6314

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests

### CODE REVIEW

File: src/tests/markdown.spec.js [hunk 1]
```diff
@@ -123,8 +123,9 @@ describe('Markdown though editor', () => {
 		expect(markdownThroughEditor('---\n---')).toBe('---\n---')
 	})
 
-	test('mentions', () => {
-		expect(markdownThroughEditor('@[username](mention://user/id)')).toBe(' @[username](mention://user/id) ')
+	test.only('mentions', () => {
```
Comment (max-nextcloud) [inline src/tests/markdown.spec.js:126]:
```suggestion
	test('mentions', () => {
```

File: src/tests/markdown.spec.js [hunk 2]
```diff
@@ -196,9 +197,10 @@ describe('Markdown serializer from html', () => {
 		expect(markdownThroughEditorHtml('<pre id="frontmatter"><code>---</code></pre><h1>Heading</h1>')).toBe('----\n---\n----\n\n# Heading')
 	})
 
-	test('mentions', () => {
-		expect(markdownThroughEditorHtml('<span class="mention" data-label="username" data-type="user" data-id="id">username</span>')).toBe(' @[username](mention://user/id) ')
-		expect(markdownThroughEditorHtml('<span class="mention" data-label="whitespace user" data-type="user" data-id="whitespace user">whitespace user</span>')).toBe(' @[whitespace user](mention://user/whitespace%20user) ')
+	test.only('mentions', () => {
```
Comment (max-nextcloud) [inline src/tests/markdown.spec.js:200]:
```suggestion
	test('mentions', () => {
```

**Review by max-nextcloud — CHANGES_REQUESTED** [review]
Fine with me... just needs to remove the `.only` in the tests to make sure the other tests still run.

File: src/tests/markdown.spec.js (hunk 1 above)
Comment (mejo-) [reply src/tests/markdown.spec.js]: Good catch, fixed now.

**Review by luka-nextcloud — APPROVED** [review]

**Pull request merged**